packages/core/src/components/menu/menuItem.tsx

```diff
@@ -133,32 +137,36 @@ export class MenuItem extends AbstractPureComponent<IMenuItemProps, IMenuItemSta
         );
 
         if (hasSubmenu) {
-            const measureSubmenu = this.props.useSmartPositioning ? this.measureSubmenu : null;
-            const submenuElement = <Menu ref={measureSubmenu}>{this.renderChildren()}</Menu>;
-            const popoverClasses = classNames(Classes.MINIMAL, Classes.MENU_SUBMENU, popoverProps.popoverClassName, {
+            const submenuRef = <Menu>{this.renderChildren()}</Menu>;
+            const popoverClasses = classNames(Classes.MENU_SUBMENU, popoverProps.popoverClassName, {
+                // apply this class to make the popover anchor to the *left*
+                // side of the menu (setting Position.LEFT_TOP alone would still
+                // anchor the popover to the right edge of the target).
                 [Classes.ALIGN_LEFT]: this.state.alignLeft,
             });
 
             content = (
                 <Popover
-                    isDisabled={disabled}
+                    disabled={disabled}
                     enforceFocus={false}
                     hoverCloseDelay={0}
                     inline={true}
                     interactionKind={PopoverInteractionKind.HOVER}
                     position={this.state.alignLeft ? Position.LEFT_TOP : Position.RIGHT_TOP}
-                    useSmartArrowPositioning={false}
                     {...popoverProps}
-                    content={submenuElement}
+                    content={submenuRef}
+                    minimal={true}
```

Comment:
Use the `minimal` prop instead of applying the `pt-minimal` class alone. This ensures the menu won't leave room for an arrow that doesn't even exist.